src/geode/io/mesh/vtp_input.cpp

```diff
@@ -0,0 +1,424 @@
+                absl::StrCat( "[VTPInput::decode] Fixed header size is wrong "
+                              "(should be 12 bytes, got ",
+                    bytes.size(), " bytes)" ) );
+            const auto fixed_header_values =
+                reinterpret_cast< const geode::index_t* >(
+                    bytes.c_str() ); // should be unsigned long if UInt64
+            const auto nb_data_blocks = fixed_header_values[0];
+            if( nb_data_blocks == 0 )
+            {
+                return std::vector< T >{};
+            }
+            const auto uncompressed_block_size = fixed_header_values[1];
+            const auto nb_characters =
+                std::ceil( nb_data_blocks * 32. / 24. ) * 4;
+            auto optional_header = clean_input.substr( 16, nb_characters );
+            decode_status = absl::Base64Unescape( optional_header, &bytes );
+            OPENGEODE_EXCEPTION( decode_status,
+                "[VTPInput::decode] Error in decoding base64 "
+                "data for optional header" );
+            OPENGEODE_ASSERT( bytes.size() == nb_data_blocks * 4,
+                absl::StrCat(
+                    "[VTPInput::decode] Fixed header size is wrong (should be ",
+                    nb_data_blocks * 4, " bytes, got ", bytes.size(),
+                    " bytes)" ) );
+            const auto optional_header_values =
+                reinterpret_cast< const geode::index_t* >(
+                    bytes.c_str() ); // should be unsigned long if UInt64
+            geode::index_t sum_compressed_block_size{ 0 };
+            absl::FixedArray< geode::index_t > compressed_blocks_size(
+                nb_data_blocks );
+            for( const auto b : geode::Range{ nb_data_blocks } )
+            {
+                compressed_blocks_size[b] = optional_header_values[b];
+                sum_compressed_block_size += optional_header_values[b];
+            }
+
+            const auto data_offset = 16 + nb_characters;
+            auto data = clean_input.substr(
+                data_offset, clean_input.size() - data_offset );
+            decode_status = absl::Base64Unescape( data, &bytes );
+            OPENGEODE_EXCEPTION( decode_status,
+                "[VTPInput::decode] Error in decoding base64 data" );
+            const auto compressed_data_bytes =
+                reinterpret_cast< const unsigned char* >( bytes.c_str() );
+
+            std::vector< T > result;
+            result.reserve( std::ceil(
+                nb_data_blocks * uncompressed_block_size / sizeof( T ) ) );
+            geode::index_t cur_data_offset{ 0 };
+            for( const auto b : geode::Range{ nb_data_blocks } )
+            {
+                const auto compressed_data_length =
+                    3 * sum_compressed_block_size;
+                size_t decompressed_data_length =
+                    nb_data_blocks * uncompressed_block_size;
+                std::unique_ptr< uint8_t > decompressed_data_bytes{
```

Comment:
Use a fixedArray since you know the size and the method `data()` to get the pointer.